### ISSUE TITLE
incorrect outer label in c language

### ISSUE THREAD

**Issue by sorthoth** [EsotericSoftware/spine-runtimes#1000]
https://github.com/EsotericSoftware/spine-runtimes/blob/a8f50bdeccaedffe55a792b15ca39ad6a80a3220/spine-libgdx/spine-libgdx/src/com/esotericsoftware/spine/AnimationState.java#L728

	public static void main(String[] args) {
		outer:
		for (int i = 0; i < 5; i++)
		{
			if (i == 3)
				continue outer;
			System.out.println(i);
		}
	}
/* output */
0
1
2
4

https://github.com/EsotericSoftware/spine-runtimes/blob/9e97eb2bfae379a3eea41e50301b7c896dc93413/spine-c/spine-c/src/spine/AnimationState.c#L929

	int main()
	{
		for (int i = 0; i < 5; i++)
		{
			if (i == 3)
			{
				i++;
				goto outer;
			}
			std::cout << i << std::endl;
		}
		outer:
		return 0;
	}
/* output */
0
1
2

https://github.com/EsotericSoftware/spine-runtimes/blob/a8f50bdeccaedffe55a792b15ca39ad6a80a3220/spine-csharp/src/AnimationState.cs#L722

        static void Main(string[] args)
        {
            for (int i = 0; i < 5; i++)
            {
                if (i == 3)
                    goto outer;
                Console.WriteLine(i);
                outer: { }
            }
        }
/* output */
0
1
2
4

**Comment by NathanSweet** [comment]
It looks like this was ported as `break outer` while the ref impl has `continue outer`.

We could consider label naming conventions to make the intention clear, eg `goto break_outer` and `goto continue_outer`. Then we'd always use this boilerplate (omitting a label if not needed):
```
i = 0;
continue_outer:
for (; i < n; i ++) {
	...
}
break_outer: {}
```
This could also be done in C#, @pharan.

**Comment by NathanSweet** [comment]
Alternatively:
```
for (i = 0; i < n; i ++) {
	...
	continue_outer: {}
}
break_outer: {}
```
Though, it is kind of nice to have `continue_outer` in the same place as the ref impl (before the loop). But then it's uglier to move out the for loop initializer. I'm fine with either way.

**Comment by badlogic** [comment]
@NathanSweet that would considerably make porting of this error prone construct easier. I think we can postpone "backporting" the convention for now, but do it for new code in the reference implementation. If you want it backported, let me know.

@sorthoth thanks, fixed in 3.6 and 3.7-beta branch.

**Comment by NathanSweet** [comment]
There's only maybe 5 gotos, would be nice for them to all be pretty.
